src/Elliptic/DiscontinuousGalerkin/Initialization.cpp

```diff
@@ -0,0 +1,156 @@
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace elliptic::dg {
+
+template <size_t Dim>
+void InitializeGeometry<Dim>::operator()(
+    const gsl::not_null<Mesh<Dim>*> mesh,
+    const gsl::not_null<Element<Dim>*> element,
+    const gsl::not_null<ElementMap<Dim, Frame::Inertial>*> element_map,
+    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Logical>*>
+        logical_coords,
+    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
+        inertial_coords,
+    const gsl::not_null<
+        InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
+        inv_jacobian,
+    const gsl::not_null<Scalar<DataVector>*> det_inv_jacobian,
+    const std::vector<std::array<size_t, Dim>>& initial_extents,
+    const std::vector<std::array<size_t, Dim>>& initial_refinement,
+    const Domain<Dim>& domain,
+    const ElementId<Dim>& element_id) const noexcept {
+  // Mesh
+  const auto quadrature = Spectral::Quadrature::GaussLobatto;
+  *mesh = domain::Initialization::create_initial_mesh(initial_extents,
+                                                      element_id, quadrature);
+  // Element
+  const auto& block = domain.blocks()[element_id.block_id()];
+  if (block.is_time_dependent()) {
+    ERROR_NO_TRACE(
+        "The InitializeDomain action is for elliptic systems which do not have "
+        "any time-dependence, but the domain creator has set up the domain to "
+        "have time-dependence.");
+  }
+  *element = domain::Initialization::create_initial_element(element_id, block,
+                                                            initial_refinement);
+  // Element map
+  *element_map = ElementMap<Dim, Frame::Inertial>{
+      element_id, block.stationary_map().get_clone()};
+  // Coordinates
+  *logical_coords = logical_coordinates(*mesh);
+  *inertial_coords = element_map->operator()(*logical_coords);
+  // Jacobian
+  *inv_jacobian = element_map->inv_jacobian(*logical_coords);
+  *det_inv_jacobian = determinant(*inv_jacobian);
+}
+
+template <size_t Dim>
+void InitializeFacesAndMortars<Dim>::operator()(
+    const gsl::not_null<std::unordered_set<Direction<Dim>>*>
+        internal_directions,
+    const gsl::not_null<std::unordered_set<Direction<Dim>>*>
+        external_directions,
+    const gsl::not_null<std::unordered_map<Direction<Dim>, Direction<Dim>>*>
+        face_directions_internal,
+    const gsl::not_null<
+        std::unordered_map<Direction<Dim>, tnsr::I<DataVector, Dim>>*>
+        face_inertial_coords_internal,
+    const gsl::not_null<
+        std::unordered_map<Direction<Dim>, tnsr::i<DataVector, Dim>>*>
+        face_normals_internal,
+    const gsl::not_null<std::unordered_map<Direction<Dim>, Scalar<DataVector>>*>
```

Comment:
I figured this is the cleanest way. The initializer declares the face-normal magnitudes in the `return_tags` so they get added to the DataBox. It requires running `NormalizeFaceNormal` after initializing background fields anyway, so it's an implementation detail if it needs to do anything with the magnitudes _before_ initializing background fields or not. E.g. it could initialize the direction-map with correctly sized `Scalars` at this point, then fill them later. I decided to just leave the direction-map uninitialized here and fill it when normalizing, but again this is an implementation detail. I think it's pretty clean :)